### PR TITLE
JUnit task.

### PR DESCRIPTION
**Pull request by dceaicovschi**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -19,6 +19,58 @@
...
+            <scope>test</scope>
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/org.junit.jupiter/junit-jupiter-engine -->
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>5.7.0</version>
+            <scope>test</scope>
+        </dependency>
+
```
Comment (en-rmrosca) [inline pom.xml]: delete empty lines

File: src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
+package com.endava.soa_5.JUnit_and_Hamcrest;
+
+public class MagicSquare {
+
+    private int[][] intArray;
+
+    private int constant;
+
+    public MagicSquare(int[][] intArray) {
+
```
Comment (en-rmrosca) [inline src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java]: remove empty line

File: src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
+package com.endava.soa_5.JUnit_and_Hamcrest;
+
+public class MagicSquare {
+
+    private int[][] intArray;
+
```
Comment (en-rmrosca) [inline src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java]: remove empty line

File: src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    private int constant;
```
Comment (en-rmrosca) [inline src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java]: please rename to magicSquareConstant

File: src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    public MagicSquare(int[][] intArray) {
+
+        this.intArray = intArray;
+        constant = intArray.length * (intArray.length * intArray.length + 1) / 2;
+    }
+
+    public boolean isNotEmpty() {
+        if (intArray.length == 0) {
+            return false;
+        } else {
```
Comment (en-rmrosca) [inline src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java]: else can be removed. Just return true after the if statement code block

File: src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+            return true;
+        }
+    }
+
+    public boolean isMultidimensionalArray() {
+
```
Comment (en-rmrosca) [inline src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java]: remove empty line

File: pom.xml (hunk 1 above)
Comment (dceaicovschi) [reply pom.xml]: done!

File: src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java (hunk 3 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java]: Done!

File: src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java (hunk 4 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java]: Done!

File: src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java (hunk 5 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java]: Done!

File: src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java (hunk 6 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquare.java]: Done!

**Review by margaritaq — APPROVED** [review]

File: src/test/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquareTest.java [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+        for (MagicSquare square : squareList) {
+            assertThat("Check that biDimensional array is not Empty", square.isNotEmpty(),
+                    is(true));
+        }
+    }
+
+    @Test
+    void checkIsEmptySquare() {
+        assertThat("Check that biDimensional array is Empty", emptyValue.isNotEmpty(),
+                is(false));
```
Comment (caheletavi) [inline src/test/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquareTest.java:97]: This assert is a real roller coaster... Could you please simplify it to something like: 
`assertTrue("Check that biDimensional array is Empty", emptyValue.isEmpty());`

Comment (dceaicovschi) [reply src/test/java/com/endava/soa_5/JUnit_and_Hamcrest/MagicSquareTest.java:97]: I renamed isNotEmpty into isNotEmptyArray, to be more clear what method is tested

**Pull request merged**